task_sdk/src/airflow/sdk/definitions/connection.py

```diff
@@ -50,3 +55,32 @@ class Connection:
     def get_uri(self): ...
 
     def get_hook(self): ...
+
+    def _get_extra_dejson(self, nested: bool = False) -> dict:
```

Comment:
Ported mainly from https://github.com/apache/airflow/blob/a6da8df0d7bb1958b7e3ab558e06d16ce72b5327/airflow/models/connection.py#L413 with some minor changes. 

The `nested` parameter is never used at all in legacy code. Something worth nuking and building some intelligence to check for `nested` jsons without a parameter is a direction I am favouring, thoughts @kaxil @ashb?

Reply:
I think it is planned to be used in https://github.com/apache/airflow/pull/35591/

Reply:
Given it has been long since that PR is up, I am fine for nuking it

Reply:
Yeah, i think we should do that, no activity on that one.

Reply:
Also https://github.com/apache/airflow/pull/39811 was pre maturely merged for future support for https://github.com/apache/airflow/pull/35591 which is 15 months old. There were some open conversations. I think we should nuke it.

Reply:
I nuked it in: 49950cbcbec7764477cddcf4d86351ee6a0ba155